server/server.go

```diff
@@ -3721,23 +3730,28 @@ func (s *StanServer) performAckExpirationRedelivery(sub *subState, isStartup boo
 		// otherwise this could cause a message to be redelivered to multiple members.
 		if !isClustered && qs != nil && !isStartup {
 			qs.Lock()
+			sub.Lock()
+			skip := !sub.isMsgStillPending(m)
```

Comment:
very minor nitpick, take it or leave it, but could use `msgPending := sub.isMsgStillPending(m)` and avoid the `!`s.  It might read easier.

Reply:
Will do, and actually should not try to send if not pending.. so thanks!